bin-src/tasks/snapshot.js

```diff
@@ -95,13 +96,17 @@ export const takeSnapshots = async (ctx, task) => {
     case 'BROKEN':
       setExitCode(ctx, exitCodes.BUILD_HAS_ERRORS, true);
       ctx.log.error(buildHasErrors(ctx));
-      transitionTo(buildFailed, true)(ctx, task);
+      transitionTo(buildBroken, true)(ctx, task);
       break;
 
     case 'FAILED':
-    case 'CANCELLED':
       setExitCode(ctx, exitCodes.BUILD_FAILED, true);
-      transitionTo(buildError, true)(ctx, task);
+      transitionTo(buildFailed, true)(ctx, task);
+      break;
+
+    case 'CANCELLED':
```

Comment:
Perhaps we should just handled both strings here with the assumption that we'll figure out a way to migrate the API at some stage?

Reply:
The only way I see that working is if we release patches for old versions of the CLI. If the index is all the sudden going to return `CANCELED`, old CLI versions are not going to accept that. Not that builds get canceled all that often, but still.

Reply:
Eventually we will know when older versions have been phased out we can make the switch? Alternatively we could add a second type of "legacy" field on the `Build.status` resolver. I guess that'd be the more correct thing to do?

Reply:
We actually send the CLI version (`packageVersion`) when we create the build, so we could use that to send the appropriate response.

Reply:
Let's deal with this some other time.